src/platforms/evdev/libinput_device.cpp

```diff
@@ -350,7 +350,13 @@ void mie::LibInputDevice::handle_touch_down(libinput_event_touch* touch)
 void mie::LibInputDevice::handle_touch_up(libinput_event_touch* touch)
 {
     MirTouchId const id = libinput_event_touch_get_slot(touch);
-    last_seen_properties[id].action = mir_touch_action_up;
+    //Only update this id if there is a valid record in the map. Otherwise, if an invalid 
+    //"up" event is being received from a stupid panel this will create fake actions.
+    //the entry was deleted when the proper "up" event was received, but C++ map array
+    //function will insta-create a new one otherwise
+    auto const i = last_seen_properties.find(id);
+    if (i != end(last_seen_properties))
+        last_seen_properties[id].action = mir_touch_action_up;
```

Comment:
To not look up the hash table twice, you can de-reference the returned iterator directly.

(I've not tested the change below)

```suggestion
    auto it = last_seen_properties.find(id);
    if (it != end(last_seen_properties))
        (*it).action = mir_touch_action_up;
```